core/vm/evm.go

```diff
@@ -305,7 +285,7 @@ func (evm *EVM) Call(caller ContractRef, addr common.Address, input []byte, gas
 	}
 
 	if isPrecompile {
-		ret, gas, err = RunStatefulPrecompiledContract(p, evm, caller.Address(), addr, input, gas, evm.interpreter.readOnly)
+		ret, gas, err = p(caller.Address(), input, gas, evm.interpreter.readOnly)
```

Comment:
This moves further from upstream, which uses [`RunPrecompiledContract()`](https://pkg.go.dev/github.com/ethereum/go-ethereum/core/vm#RunPrecompiledContract) to wrap tracing and gas metering.

Reply:
We are also not using `RunPrecompiledContract`, so I don't think it changes how far this is from upstream.
Also my long term plan is to propose the CustomPrecompiles to upstream as a configuration option, but it may not be accepted, so we see.
But open to suggestions.